### PR TITLE
pass in number of tokens to withdraw

### PR DESCRIPTION
**Pull request by RusseII**
image

This change is to support the withdraw panel. Previously it was only possible to withdraw 100% of the available collateral. 

This method will also be called in the future for refinancing.

### CODE REVIEW

**Review by Namaskar-1F64F — APPROVED** [review]
This looks good 👍 

Should we move "withdrawExcessPayment" under pay?

File: contracts/Bond.sol [hunk 1]
```diff
@@ -111,21 +111,20 @@ contract Bond is
...
     /// @inheritdoc IBond
-
-    function withdrawExcessCollateral(address receiver) external onlyOwner {
-        uint256 collateralToSend = previewWithdraw(0);
+    function withdrawExcessCollateral(uint256 amount, address receiver)
+        external
+        onlyOwner
+    {
+        if (amount > previewWithdraw(0)) {
+            revert NotEnoughCollateral();
```
[the remark this answers is not in the document]
Comment (Pet3ris) [reply contracts/Bond.sol:119]: I think reverting is fine given the collateral is not going anywhere.

File: test/Bond.spec.ts [hunk 2]
```diff
@@ -405,6 +405,36 @@ describe.only("Bond", () => {
...
+                    (await bond.previewWithdraw(0)).div(2),
+                    owner.address
+                  ),
+                collateralToken,
+                owner,
+                bond.address,
+                config.collateralTokenAmount.div(2)
+              );
+            });
+            it.only("fails if requesting too much", async () => {
```
Comment (Namaskar-1F64F) [inline test/Bond.spec.ts:425]: remove the `only` I would suggest using the 
`npx hardhat test --grep "requesting too"` 
commands. https://ethereum.stackexchange.com/a/123403/93861

**Comment by RusseII** [comment]
> Should we move "withdrawExcessPayment" under pay?

Yeah ... we can probably reorder a few of the methods

**Comment by Namaskar-1F64F** [comment]
> > Should we move "withdrawExcessPayment" under pay?
> 
> Yeah ... we can probably reorder a few of the methods

If you end up doing this - keep the style guide in mind for function grouping 👍 https://docs.soliditylang.org/en/v0.8.13/style-guide.html#order-of-functions

File: test/Bond.spec.ts (hunk 2 above)
Comment (RusseII) [reply test/Bond.spec.ts:425]: That's cool. I usually use the mocha extension though. Removed

**Pull request merged**